### PR TITLE
RustCheck: new crates checks

### PR DESCRIPTION
**Pull request by arthurzam**
Resolves: https://github.com/pkgcore/pkgcheck/issues/586

### CODE REVIEW

File: src/pkgcheck/checks/rust.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def feed(self, pkg: bash.ParseTree):
+        if "cargo" not in pkg.inherited:
+            return
+        for node in pkg.global_query(bash.var_assign_query):
+            name = pkg.node_str(node.child_by_field_name("name"))
+            if name == "CRATES":
+                val_node = node.children[-1]
+                row, _ = val_node.start_point
+                val_str: str = pkg.node_str(val_node).strip("'\"")
+                for lineno, line in enumerate(val_str.splitlines(), start=row + 1):
```
Comment (mgorny) [inline src/pkgcheck/checks/rust.py:54]: This variable is just whitespace separated, so technically speaking there could be multiple tokens per line.

File: src/pkgcheck/checks/rust.py [hunk 2]
```diff
@@ -35,7 +50,7 @@ def feed(self, pkg: bash.ParseTree):
             if name == "CRATES":
                 val_node = node.children[-1]
                 row, _ = val_node.start_point
-                val_str: str = pkg.node_str(val_node).strip("'\"")
```
Comment (mgorny) [inline src/pkgcheck/checks/rust.py]: This change seems to be on the wrong commit.

File: src/pkgcheck/checks/rust.py [hunk 3]
```diff
@@ -45,3 +60,21 @@ def feed(self, pkg: bash.ParseTree):
...
+            call_name = pkg.node_str(node.child_by_field_name("name"))
+            if call_name == "cargo_crate_uris":
+                row, _ = node.start_point
+                line = pkg.node_str(node.parent)
+                if node.child_count == 1 or (
+                    node.child_count == 2
+                    and any(
+                        pkg.node_str(var_node) == "CRATES"
+                        for var_node, _ in bash.var_query.captures(node.children[1])
+                    )
```
Comment (mgorny) [inline src/pkgcheck/checks/rust.py:73]: I'm not sure if I understand this correctly but the replacement works only if `${CRATES}` (or `"${CRATES}"`) is the only argument.

Comment (arthurzam) [reply src/pkgcheck/checks/rust.py:73]: Yes, that is why I verify the node count is exactly 2. I've even added a test for `${CRATES} something` and verify it ignores it :)

File: src/pkgcheck/checks/rust.py (hunk 1 above)
Comment (arthurzam) [reply src/pkgcheck/checks/rust.py:54]: True, maybe just use split... will see

Comment (arthurzam) [reply src/pkgcheck/checks/rust.py:52]: fixed

File: src/pkgcheck/checks/rust.py (hunk 3 above)
Comment (mgorny) [reply src/pkgcheck/checks/rust.py:73]: Well, I guess I don't understand what's the iteration for but I guess it's high magic.

**Pull request merged**